### DISCUSSION TITLE
Does time stand still during testing?

### DISCUSSION THREAD

**Discussion by mdantonio** [cypress-io/cypress#15530] [Questions and Help] [answered]
After noticed that random values are always the same (I were initially thinking about a seeding problem) and produced TOTP were invalid after 30 seconds I did a simple test:

```
describe("Test date", () => {

    it("Test date", () => {

        cy.log(new Date().toString());

        cy.wait(5000);

        cy.log(new Date().toString());

    });
});
```

I was expecting that the second that was exactly 5 seconds after the first... but they are exactly the same... so the time is not going during tests.

After reading https://docs.cypress.io/api/commands/clock.html I'm still confused. I don't want to manually sync the time, I want a normal time flow. What I'm missing?

Thank you for any help!

**Comment by herecydev** [comment, marked as answer]
Try this

```
describe("Test date", () => {
    it("Test date", () => {
        cy.log(new Date().toString());

        cy.wait(5000).then(() => {
          cy.log(new Date().toString());
        });
    });
});
```

The queueing of work can be a little unexpected if you're not used to it; it's [documented here](https://docs.cypress.io/guides/core-concepts/introduction-to-cypress.html#Commands-Are-Asynchronous). Effectively all of your code is evaluated in that test and then queued up so the time is appearing the same (even though the wait *did* happen). By using the `then` command, you are chaining the second `log` command to wait for the finish of the `wait`

**Reply by mdantonio** [reply]
It makes sense... thank you!
